analytic_engine/src/sst/parquet/writer.rs

```diff
@@ -227,40 +262,47 @@ impl<'a> SstWriter for ParquetSstWriter<'a> {
         &mut self,
         request_id: RequestId,
         meta: &MetaData,
-        record_stream: RecordBatchStream,
+        input: RecordBatchStream,
     ) -> writer::Result<SstInfo> {
         debug!(
             "Build parquet file, request_id:{}, meta:{:?}, num_rows_per_row_group:{}",
             request_id, meta, self.num_rows_per_row_group
         );
 
-        let total_row_num = Arc::new(AtomicUsize::new(0));
-        let reader = RecordBytesReader {
+        let group_writer = RecordBatchGroupWriter {
             hybrid_encoding: self.hybrid_encoding,
             request_id,
-            record_stream,
+            input,
             num_rows_per_row_group: self.num_rows_per_row_group,
+            max_buffer_size: self.max_buffer_size,
             compression: self.compression,
-            total_row_num: total_row_num.clone(),
             meta_data: meta.clone(),
-            partitioned_record_batch: Default::default(),
         };
-        let bytes = reader.read_all().await?;
-        self.store
-            .put(self.path, bytes.into())
-            .await
-            .context(Storage)?;
 
-        let file_head = self.store.head(self.path).await.context(Storage)?;
+        let (aborter, sink) =
+            ObjectStoreMultiUploadAborter::initialize_upload(self.store, self.path).await?;
+        let total_num_rows = match group_writer.write_all(sink).await {
+            Ok(v) => v,
+            Err(e) => {
+                if let Err(e) = aborter.abort().await {
```

Comment:
Leave a comment that this may leave temp file in underlying storage, and we may need to provide a repair command to clean those.